docs/recipes/mpi.md

```diff
@@ -0,0 +1,213 @@
+
+Here is an example script (e.g. named `job.sh`) to run an MPI job using multiple cores on a single node. 
+```
+#!/bin/bash
+#SBATCH -p sched_mit_hill
+#SBATCH -t 30
+#SBATCH -N 1
+#SBATCH -n 8
+#SBATCH --mem=10GB   
+
+srun hostname
+module load gcc/6.2.0 openmpi/3.0.4
+mpirun -n $SLURM_NTASKS my_program
+```
+
+This job requests 8 cores with `-n` and 10 GB of memory with `--mem` on 1 node (specified with `-N`). The `-n` flag is the same as `--ntasks`. The specified value is saved to the variable `SLURM_NTASKS`. In this case, the requested number of cores is equal to `SLURM_NTASKS`. The command `mpirun -n $SLURM_NTASKS` ensures that each MPI task runs on one core. 
+
+The command `srun hostname` is to check if the correct number of cores and nodes are assigned to the job. It is not needed in production runs. 
+
+??? "Side note: partitions and modules"
+    The modules used in the above example is for the CentOS 7 OS, which works for these partitions: `sched_mit_hill`, `newnodes`, and `sched_any`. If using a partition with the Rocky 8 OS, such as `sched_mit_orcd`, change the modules accrodingly (see the first session). 
+
+Submit the job with the `sbatch` command,
+```
+sbatch job.sh
+```
+
+To run an MPI job on multiple nodes, refer to the following exmaple script.
+```
+#!/bin/bash
+#SBATCH -p sched_mit_hill
+#SBATCH -t 30
+#SBATCH -N 2
+#SBATCH --ntasks-per-node=8
+#SBATCH --mem=10GB
+
+srun hostname
+module load gcc/6.2.0 openmpi/3.0.4
+mpirun -n $SLURM_NTASKS my_program
+```
+
+This job requests 2 nodes with 8 cores and 10 GB of memory per node. In this case, the total number of cores (saved to `SLURM_NTASKS`) is equal to the number of nodes (saved to `SLURM_NNODES`) times the number of cores per node (saved to `SLURM_NTASKS_PER_NODE`). The command `mpirun -n $SLURM_NTASKS` ensures that MPI tasks are distributed to both nodes and each MPI task runs on one core. 
+
+Alternatively, users can specify the number of cores per node using an OpenMPI option like this `mpirun -npernode $SLURM_NTASKS_PER_NODE my_program`.
+
+If replacing `--ntasks-per-node=8` with `-n 16` in the above script, the job will request 16 cores on 2 nodes, but it is not always the case that there are 8 cores per node. For example, there may be 7 cores on one node and 9 cores on another, or 1 core on one node and 15 cores on another, etc, depending on the current available resources on the cluster. 
+
+
+## Computing resources for MPI jobs
+
+To get a better idea on how many nodes, cores and memory should be requested, users need to consider the following two questions. 
+
+First, what resources are available on the cluster? Use this command to check node and job info on the cluster, including the constraint associated with OS (`%f`), the nubmer of CPU cores (`%c`), the memory size (`%m`), the wall time limit (`%l`), and the current usage status (`%t`). 
+```
+ sinfo -N -p sched_mit_hill,newnodes,sched_any,sched_mit_orcd -o %f,%c,%m,%l,%t |grep -v drain
```

Comment:
I suggest leaving out the -p flag, by default users will only see partitions they can submit jobs to, and adding %P to the output so users know which partition to select:
` sinfo -N -o %P,%f,%c,%m,%l,%t |grep -v drain`